File: src/main/java/gg/mineral/practice/util/math/PearlCooldown.java
package gg.mineral.practice.util.math;

import org.bukkit.Bukkit;

import gg.mineral.practice.PracticePlugin;
import gg.mineral.practice.entity.Profile;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
@Getter
public class PearlCooldown {
	private final Profile profile;
	private int timeRemaining = 0;

	public void start() {
		Bukkit.getServer().getScheduler().scheduleSyncRepeatingTask(PracticePlugin.INSTANCE, () -> {
			if (timeRemaining > 0)
				profile.getPlayer().setLevel(--timeRemaining);
		}, 0L, 20L);
	}

	public boolean isActive() {
		return timeRemaining > 0;
	}

	public void setTimeRemaining(int time) {
		this.timeRemaining = time;
		profile.getPlayer().setLevel(time);
	}

}

File: src/main/kotlin/gg/mineral/practice/util/math/PearlCooldown.kt
package gg.mineral.practice.util.math

import gg.mineral.practice.PracticePlugin
import it.unimi.dsi.fastutil.objects.Object2IntOpenHashMap
import org.bukkit.Bukkit
import java.util.*

class PearlCooldown {
    val cooldowns = Object2IntOpenHashMap<UUID>()

    init {
        Bukkit.getServer().scheduler.scheduleSyncRepeatingTask(
            PracticePlugin.INSTANCE,
            {
                for (entry in cooldowns.object2IntEntrySet()) {
                    val uuid = entry.key
                    var cooldown = entry.intValue

                    if (cooldown > 0) {
                        cooldowns.put(uuid, --cooldown)
                        val player = Bukkit.getPlayer(uuid)
                        player.level = cooldown
                    }
                }
            }, 0L, 20L
        )
    }

    fun isActive(uuid: UUID) = cooldowns.getInt(uuid) > 0

    fun getTimeRemaining(uuid: UUID) = cooldowns.getInt(uuid)
}
